internal/api/oidc/integration_test/token_exchange_test.go

```diff
@@ -25,40 +25,71 @@ import (
 	"github.com/zitadel/zitadel/pkg/grpc/feature/v2"
 )
 
-func setTokenExchangeFeature(t *testing.T, value bool) {
-	iamCTX := Instance.WithAuthorization(CTX, integration.UserTypeIAMOwner)
+func setTokenExchangeFeature(t *testing.T, instance *integration.Instance, value bool) {
+	iamCTX := instance.WithAuthorization(CTX, integration.UserTypeIAMOwner)
 
-	_, err := Instance.Client.FeatureV2.SetInstanceFeatures(iamCTX, &feature.SetInstanceFeaturesRequest{
+	_, err := instance.Client.FeatureV2.SetInstanceFeatures(iamCTX, &feature.SetInstanceFeaturesRequest{
 		OidcTokenExchange: proto.Bool(value),
 	})
 	require.NoError(t, err)
+	retryDuration := time.Minute
+	if ctxDeadline, ok := iamCTX.Deadline(); ok {
+		retryDuration = time.Until(ctxDeadline)
+	}
+	require.EventuallyWithT(t,
+		func(ttt *assert.CollectT) {
+			f, err := instance.Client.FeatureV2.GetInstanceFeatures(iamCTX, &feature.GetInstanceFeaturesRequest{
+				Inheritance: true,
+			})
+			assert.NoError(ttt, err)
+			if f.OidcTokenExchange.GetEnabled() {
+				return
+			}
```

Comment:
As the API is strongly consistent (`GetInstanceFeatures` uses eventstore) and the instance middleware is eventually consitent (uses projections), the acutal test can still fail on the feature flag being disabled.

I beleive we had a call about this, and we concluded that it would be better to call the token exchange with some placeholder data (invalid token) and assert the actual error not being a feature releated error.